pkg/cmd/account/aws/list/list.go

```diff
@@ -0,0 +1,72 @@
+package list
+
+import (
+	"github.com/MakeNowJust/heredoc/v2"
+	"github.com/OctopusDeploy/cli/pkg/factory"
+	"github.com/OctopusDeploy/cli/pkg/output"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/accounts"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/client"
+	"github.com/briandowns/spinner"
+	"github.com/spf13/cobra"
+)
+
+func NewCmdList(f factory.Factory) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List aws accounts",
+		Long:  "List aws accounts in an instance of Octopus Deploy.",
+		Example: heredoc.Doc(`
+			$ octopus account aws list"
+		`),
+		Aliases: []string{"ls"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := f.GetSpacedClient()
+			if err != nil {
+				return err
+			}
+			return listAwsAccounts(client, cmd, f.Spinner())
+		},
+	}
+
+	return cmd
+}
+
+func listAwsAccounts(client *client.Client, cmd *cobra.Command, s *spinner.Spinner) error {
+	s.Start()
+	accountResources, err := client.Accounts.Get(accounts.AccountsQuery{
+		AccountType: accounts.AccountTypeAmazonWebServicesAccount,
+	})
+	if err != nil {
+		return err
+	}
+	items, err := accountResources.GetAllPages(client.Accounts.GetClient())
+	if err != nil {
+		return err
+	}
+	s.Stop()
```

Comment:
do you need to `defer` this? If we return early with an error will the spinner just keep running, or is there some wrapper which handles it externally?

Reply:
good catch with the spinner, if the program exists the spinner stops and cleans itself. So errors work fine. BUT to be safe I added some stops before the returns on the errors as it does feel safer. Sadly you can not use defer on spinner stop, it messes up leaves the spinner before the start of the printed text and the terminal in raw mode after the program exits